content/blog/bigdata-boutique-guest-post/index.md

```diff
@@ -0,0 +1,66 @@
+
+To support our customers and help them succeed in their Big Data projects, we provide an array of advanced services, and our unique cost-optimization service addresses this challenge directly. This article outlines how we use long-running benchmarks to correctly identify the right configuration for our customers’ BigData clusters by using the [infrastructure as code tool Pulumi]({{< relref "/docs" >}}) for automating the creation of complex infrastructures with conditionals and highly tunable configurations to orchestrate those benchmarks at scale.
+
+## Elasticsearch Cost Optimization
+
+Let’s take Elasticsearch as an example. [Elasticsearch](https://www.elastic.co/elasticsearch) is a real-time search and analytics engine. It is built to ingest huge amounts of data and to allow one to search through and visualize them in real-time. But to be able to sustain the required amount of data on ingestion and the desired number of queries per second, the right hardware has to be used.
+
+To do this, we have to make some decisions: which disks to use (SSD disks? local, or network storage?), the number of servers (scaling up or out?), CPU and RAM ratios, index mappings, various cluster configurations, and more. These are not easy questions to answer, and that is where our [Elasticsearch expertise](https://bigdataboutique.com/services/elasticsearch) comes into play. But answering those questions correctly with a high level of certainty also requires rigorous testing for validating assumptions.
+
+Whether our customer is running on-prem or on a cloud, our [Elasticsearch Capacity Planning Service](https://bigdataboutique.com/services/elasticsearch/capacity-planning) exists for one purpose: to find the hardware solution with the optimum balance between cost and performance, and do it scientifically, so the answers are as accurate and as precise as possible.
+
+## Elasticsearch Sizing Process
+
+Our process begins with a sample data request. We ask the customer for their data (or a large sample of it), existing index mappings, expected queries, and finally known KPIs and SLAs. Our goal in the initial discovery phase is to reduce the number of free variables as much as possible because the more such variables, the more permutations we will have to test. Ideally, we want very few dimensions of freedom—usually in terms of hardware and configurations—so the number of permutations to test is on the order of dozens and not thousands.
+
+Once the initial framework has been decided, we run fully automated benchmarks to test some of the permutations our experience tells us are the most promising to deliver good results. The benchmarking process is smart enough to omit long tales and outliers, so the results are not going to lie as to the true nature of the configuration that was selected and tested.
```

Comment:
Original has this also, but happened to notice it:

```suggestion
Once the initial framework has been decided, we run fully automated benchmarks to test some of the permutations our experience tells us are the most promising to deliver good results. The benchmarking process is smart enough to omit long tails and outliers, so the results are not going to lie as to the true nature of the configuration that was selected and tested.
```

Reply:
fixed